### PR TITLE
drivers: adc: add max1125x driver

### PR DESCRIPTION
**Pull request by mustafaabdullahk**
Hi everyone !

This adds support for the max1125x (max11254, max11254) family of spi adc devices.

Signed-off-by: Mustafa Abdullah Kus <mustafa.kus@sparsetechnology.com>

### CODE REVIEW

**Comment by mustafaabdullahk** [comment]
@nashif @galak @anangl ping

File: dts/bindings/adc/maxim,max1125x-base.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Common fields for Texas Instruments MAX1125x
```
Comment (galak) [inline dts/bindings/adc/maxim,max1125x-base.yaml:1]: Is Texas Instruments correct here?

Comment (mustafaabdullahk) [reply dts/bindings/adc/maxim,max1125x-base.yaml:1]: No, incorrect. i will replace with 'Analog Devices Inc.'

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thank you for your contribution!

File: drivers/adc/adc_max1125x.c [hunk 2]
```diff
@@ -0,0 +1,871 @@
...
+	int ret;
+	const struct max1125x_config *config = dev->config;
+	uint8_t command = MAX1125X_CMD_READ | reg_addr;
+
+	const struct spi_buf tx_buf = {.buf = &command, .len = sizeof(command)};
+	const struct spi_buf_set tx = {.buffers = &tx_buf, .count = 1};
+	const struct spi_buf rx_buf = {.buf = buffer, .len = reg_size};
+	const struct spi_buf_set rx = {.buffers = &rx_buf, .count = 1};
+
+	gpio_pin_set_dt(&config->csb_gpio, 1);
```
[the remark this answers is not in the document]
Comment (mustafaabdullahk) [reply drivers/adc/adc_max1125x.c:242]: I know, I'm tried to implement it with ''struct spi_cs_control'' but the chip is not compatible with spi timing requirements.

File: drivers/adc/adc_max1125x.c [hunk 3]
```diff
@@ -0,0 +1,871 @@
...
+		odr_delay_us = odr_delay[MAX1125X_CONFIG_RATE_16000];
+		break;
+	case MAX1125X_CONFIG_RATE_32000:
+		odr = MAX1125X_CONFIG_RATE_32000;
+		odr_delay_us = odr_delay[MAX1125X_CONFIG_RATE_32000];
+		break;
+	case MAX1125X_CONFIG_RATE_64000:
+		odr = MAX1125X_CONFIG_RATE_64000;
+		odr_delay_us = odr_delay[MAX1125X_CONFIG_RATE_64000];
+		break;
```
[the remark this answers is not in the document]
Comment (mustafaabdullahk) [reply drivers/adc/adc_max1125x.c:383]: yep, you are right. i will refactor this part.

**Comment by mustafaabdullahk** [comment]
> Thank you for your contribution!



> MaureenHelm

:star_struck:

File: drivers/adc/adc_max1125x.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (MaureenHelm) [reply drivers/adc/adc_max1125x.c:242]: Which timing requirement exactly? Is this perhaps an issue with the SPI driver not supporting CS delay? Which SPI driver are you using?

File: drivers/adc/adc_max1125x.c [hunk 4]
```diff
@@ -0,0 +1,811 @@
...
+	return odr;
+}
+
+static int max1125x_wait_data_ready(const struct device *dev)
+{
+	struct max1125x_data *data = dev->data;
+
+	k_sleep(data->ready_time);
+
+	uint8_t is_ready
```
Comment (MaureenHelm) [inline drivers/adc/adc_max1125x.c:340]:
```suggestion
	uint8_t is_ready;
```
This leads me to believe this driver isn't getting built in CI. Have you tried building `tests/drivers/build_all/adc` locally?

Comment (mustafaabdullahk) [reply drivers/adc/adc_max1125x.c]: yes i run tested before pushed.

image

File: drivers/adc/adc_max1125x.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (mustafaabdullahk) [reply drivers/adc/adc_max1125x.c:242]: image

i use stm32 spi driver. i see another adc drivers communicating with same way. for e.g

image
adc_ads114s0x.c source file

File: dts/bindings/adc/maxim,max1125x-base.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    enum:
+      - 0
+      - 1
+    description: this option for gpio0 direction 1 for input, 0 for output
+  gpio1-direction:
+    type: int
+    enum:
+      - 0
+      - 1
+    description: this option for gpio0 direction 1 for input, 0 for output
```
Comment (MaureenHelm) [inline dts/bindings/adc/maxim,max1125x-base.yaml:25]: These should be booleans

Comment (mustafaabdullahk) [reply dts/bindings/adc/maxim,max1125x-base.yaml]: update as boolean

**Review by MaureenHelm — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**